src/app/template/shared/base-template.model.ts

```diff
@@ -1,74 +1,59 @@
-import * as moment from 'moment';
-
-import { FieldMapper } from '../../shared/decorators/field-mapper.decorator';
-import { BaseModel, Tag } from '../../shared/models';
+import { BaseModelInterface, Tag } from '../../shared/models';
 import { OsType } from '../../shared/models/os-type.model';
 import { Taggable } from '../../shared/interfaces/taggable.interface';
 import { TemplateTagKeys } from '../../shared/services/tags/template-tag-keys';
 import { Utils } from '../../shared/services/utils/utils.service';
 
+export enum TemplateResourceType {
+  Iso = 'Iso',
+  Template = 'Template'
+}
 
-@FieldMapper({
-  displaytext: 'displayText',
-  domainid: 'domainId',
-  isdynamicallyscalable: 'isDynamicallyScalable',
-  isextractable: 'isExtractable',
-  isfeatured: 'isFeatured',
-  ispublic: 'isPublic',
-  isready: 'isReady',
-  ostypeid: 'osTypeId',
-  ostypename: 'osTypeName',
-  zoneid: 'zoneId',
-  zonename: 'zoneName',
-})
-export abstract class BaseTemplateModel extends BaseModel implements Taggable {
-  public abstract resourceType: string;
-
-  public path: string;
-
-  public id: string;
-  public account: string;
-  public created: Date;
-  public crossZones: boolean;
-  public displayText: string;
-  public domain: string;
-  public domainId: string;
-  public isDynamicallyScalable: boolean;
-  public isExtractable: boolean;
-  public isFeatured: boolean;
-  public isPublic: boolean;
-  public isReady: boolean;
-  public name: string;
-  public osTypeId: string;
-  public osTypeName: string;
-  public osType: OsType;
-  public size: number;
-  public status: string;
-  public tags: Array<Tag>;
-  public zoneId: string;
-  public zoneName: string;
-  public agreementAccepted?: boolean;
+export interface BaseTemplateModel extends BaseModelInterface, Taggable {
```

Comment:
Can you inspect taggable interface? Why it duplicate id property and allow to add any property to object? Can we remove these properties?